tests/vmi_networking_test.go

```diff
@@ -721,43 +721,46 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			serverVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(serverVMI.Name, &v13.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			Expect(len(serverVMI.Status.Interfaces)).To(Equal(1))
-
-			serverIP := serverVMI.Status.Interfaces[0].IP
-			if netutils.IsIPv6String(serverIP) {
-				By("Checking traceroute from vmi to cluster nodes gateway")
-				// Cluster nodes subnet (docker network gateway)
-				// Docker network subnet cidr definition https://bit.ly/2wZTgMK
-				err := tests.CheckForTextExpecter(serverVMI, createExpectTraceroute6("2001:db8:1::1"), 30)
-				Expect(err).ToNot(HaveOccurred(), "Failed to traceroute to VMI %s within the given timeout", serverVMI.Name)
-			} else {
-				if ipv4NetworkCIDR == "" {
-					ipv4NetworkCIDR = api.DefaultVMCIDR
-				}
-				By("Checking ping to gateway")
-				ipAddr := gatewayIPFromCIDR(ipv4NetworkCIDR)
-				Expect(tests.PingFromVMConsole(serverVMI, ipAddr)).To(Succeed())
-
-				By("Checking ping to google")
-				Expect(tests.PingFromVMConsole(serverVMI, "8.8.8.8")).To(Succeed())
-				Expect(tests.PingFromVMConsole(clientVMI, "google.com")).To(Succeed())
-			}
+			Expect(serverVMI.Status.Interfaces[0].IPs).NotTo(BeEmpty())
 
 			By("starting a tcp server")
 			tcpPort := "8080"
 			err = tests.CheckForTextExpecter(serverVMI, createExpectStartTcpServer(tcpPort), 30)
 			Expect(err).ToNot(HaveOccurred())
 
-			By("Connecting from the client vm")
-			err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, tcpPort, true), 30)
-			Expect(err).ToNot(HaveOccurred())
+			for _, serverIP := range serverVMI.Status.Interfaces[0].IPs {
+				if netutils.IsIPv6String(serverIP) {
+					By("Testing IPv6")
+					By("Checking traceroute (IPv6) from vmi to cluster nodes gateway")
```

Comment:
Having all the details in one line before the tests start is much more easier to track instead of searching in the Expecter log

Reply:
Just one ip there is constant, the rest are dynamic and will be exactly near the command that was actually failed.
I dont see a reason to add a code which is redundant (or not needed in case of the dynamic) and to maintain it.

Reply:
You dont realy need to maintain it, it is formatted string with two variable you already have, if you insist then just leave it as is

Reply:
thx i prefer to leave it as is.